hpy/devel/include/hpy/runtime/structseq.h

```diff
@@ -0,0 +1,190 @@
+#ifndef HPY_COMMON_RUNTIME_STRUCTSEQ_H
+#define HPY_COMMON_RUNTIME_STRUCTSEQ_H
+
+#include "hpy.h"
+
+/*
+ * Struct sequences are relatively simple subclasses of tuple, so we provide the
+ * simplified API to create them here. This maps closely to the CPython limited
+ * API for creating struct sequences. However, in universal mode we use the
+ * collections.namedtuple type to implement this, which behaves a bit
+ * differently w.r.t. hidden elements. Thus, the n_in_sequence field available
+ * in CPython's PyStructSequence_Desc is not available. Also, we use a builder
+ * API like for tuples and lists so that the struct sequence is guaranteed not
+ * to be written after it is created.
+ */
+
+/**
+ * Describes a field of a struct sequence.
+ */
+typedef struct {
+    /**
+     * Name (UTF-8 encoded) for the field or ``NULL`` to end the list of named
+     * fields. Set the name to :c:var:`HPyStructSequence_UnnamedField` to leave
+     * it unnamed.
+     */
+    const char *name;
+
+    /**
+     * Docstring of the field (UTF-8 encoded); may be ``NULL``.
+     */
+    const char *doc;
+} HPyStructSequence_Field;
+
+/**
+ * Contains the meta information of a struct sequence type to create.
+ * Struct sequences are subclasses of tuple. All fields are Python objects. The
```

Comment:
Is this a copy-paste or did you really mean that the fields are Python objects?

Reply:
Right, I've basically copied that from the CPython docs.
If the phrase confuses you, I think we can just remove it. As in the same line stated: a struct sequence is a subclass of tuple. So it should be clear what are the elements of it.